althea_kernel_interface/src/open_tunnel.rs

```diff
@@ -58,13 +62,21 @@ impl KernelInterface {
         private_key_path: &Path,
         own_ip: &IpAddr,
         external_nic: Option<String>,
-        conf_link_local: bool,
+        settings_default_route: &mut Vec<String>
     ) -> Result<(), Error> {
+        let mut external_peer;
+
         let phy_name = match self.get_device_name(endpoint.ip()) {
-            Ok(phy_name) => phy_name,
-            Err(err) => external_nic.unwrap_or("no_nic".to_string()),
+            Ok(phy_name) => {
+                external_peer = false;
+                Some(phy_name)
+            },
+            Err(err) => {
+                external_peer = true;
+                external_nic
+            },
         };
-        let socket_connect_str = socket_to_string(endpoint, phy_name);
+        let socket_connect_str = socket_to_string(endpoint, phy_name.clone());
```

Comment:
why the `clone` if `phy_name` isn't used anywhere after this statement and `socket_to_string` doesn't mutate it?